src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1364,4 +1364,90 @@ impl Cage {
         }
     }
 
+    //------------------RMDIR SYSCALL------------------
+
+    pub fn rmdir_syscall(&self, path: &str) -> i32 {
+        if path.len() == 0 {return syscall_error(Errno::ENOENT, "rmdir", "Given path is null");}
+
+        let truepath = normpath(convpath(path), self);
+        let mut metadata = FS_METADATA.write().unwrap();
+
+        // try to get inodenum of input path and its parent
+        match metawalkandparent(truepath.as_path(), Some(&metadata)) {
+            (None, ..) => {
+                syscall_error(Errno::EEXIST, "rmdir", "Path does not exist")
+            }
+            (Some(_), None) => { // path exists but parent does not => path is root dir
+                syscall_error(Errno::EBUSY, "rmdir", "Cannot remove root directory")
+            }
+            (Some(inodenum), Some(parent_inodenum)) => {
+                let inodeobj = metadata.inodetable.get_mut(&inodenum).unwrap();
+
+                match inodeobj {
+                    // make sure inode matches a directory
+                    Inode::Dir(dir_obj) => {
+                        if dir_obj.linkcount > 2 {return syscall_error(Errno::EEXIST, "rmdir", "Directory is not empty");}
+                        if !is_dir(dir_obj.mode) {return syscall_error(Errno::ENOTDIR, "rmdir", "Mode is not set to directory");}
```

Comment:
This maybe should be a panic, as Inode::Dir must be a directory. @rennergade thoughts?

Reply:
Either that or don't have this check at all possibly

Reply:
Hmm. Yeah, I do think the check is worth having here. But I think I agree having it panic instead of return an error here makes more sense because if we have a directory with the mode incorrectly set then something is majorly wrong.